pkg/apis/core/validation/validation.go

```diff
@@ -4197,12 +4236,47 @@ func ValidateNode(node *core.Node) field.ErrorList {
 	// That said, if specified, we need to ensure they are valid.
 	allErrs = append(allErrs, ValidateNodeResources(node)...)
 
-	if len(node.Spec.PodCIDR) != 0 {
-		_, err := ValidateCIDR(node.Spec.PodCIDR)
+	podCIDRsField := field.NewPath("spec", "podCIDRs")
+	// validate PodCIDRS only if we need to
+	if len(node.Spec.PodCIDRs) == 0 {
+		return allErrs
```

Comment:
Don't return here - too hard to tell if there's more work to do. Either write as 

```
if len > 0 { 
  // validate code here
}
```

or call a function:

```
if len > 0 {
  allErrs = append(allErrs, validateNodePodCIDRs(node.Spec.PodCIDRs, field.NewPath("spec", "podCIDR"))...)
}
```